pkg/crc/machine/kubeconfig_test.go

```diff
@@ -59,3 +61,63 @@ func TestUpdateUserCaAndKeyToKubeconfig(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, "dummycert", userClientCA)
 }
+
+func createTempKubeConfig(config *api.Config) (string, error) {
+	tempFile, err := os.CreateTemp("", "kubeconfig-")
```

Comment:
https://pkg.go.dev/testing#T.TempDir is convenient in tests, no need to check for errors and all its content is automatically cleaned up after the test is done.